src/bio/bio_context.c

```diff
@@ -121,67 +200,166 @@ blob_wait_completion(struct bio_xs_context *xs_ctxt, struct blob_cp_arg *ba)
 	}
 }
 
-struct blob_msg_arg {
-	struct spdk_blob_opts	 bma_opts;
-	struct spdk_blob_store	*bma_bs;
-	struct spdk_blob	*bma_blob;
-	spdk_blob_id		 bma_blob_id;
-	void			*bma_cp_arg;
-};
-
 static void
 blob_msg_create(void *msg_arg)
 {
 	struct blob_msg_arg *arg = msg_arg;
 
 	spdk_bs_create_blob_ext(arg->bma_bs, &arg->bma_opts, blob_create_cb,
-				arg->bma_cp_arg);
+				msg_arg);
 }
 
 static void
 blob_msg_delete(void *msg_arg)
 {
 	struct blob_msg_arg *arg = msg_arg;
 
-	spdk_bs_delete_blob(arg->bma_bs, arg->bma_blob_id, blob_cb,
-			    arg->bma_cp_arg);
+	spdk_bs_delete_blob(arg->bma_bs, arg->bma_blob_id, blob_cb, msg_arg);
 }
 
 static void
 blob_msg_open(void *msg_arg)
 {
 	struct blob_msg_arg *arg = msg_arg;
 
-	spdk_bs_open_blob(arg->bma_bs, arg->bma_blob_id, blob_open_cb,
-			  arg->bma_cp_arg);
+	spdk_bs_open_blob(arg->bma_bs, arg->bma_blob_id, blob_open_cb, msg_arg);
 }
 
 static void
 blob_msg_close(void *msg_arg)
 {
-	struct blob_msg_arg *arg = msg_arg;
+	struct blob_msg_arg	*arg = msg_arg;
+	struct spdk_blob	*blob = arg->bma_ioc->bic_blob;
+
+	spdk_blob_close(blob, blob_close_cb, msg_arg);
+}
+
+static void
+bio_bs_unhold(struct bio_blobstore *bbs)
```

Comment:
or bio_bs_release()

Reply:
I actually deliberately avoid using bs_acquire/grab() and bs_release() for this, because it's easy to tell by function name that bs_hold()/unhold() will operate on 'bs_holdings', that could probably helpful for differentiating it from the existing bs_get() and bs_put() which operates on reference.